### PR TITLE
Change license to Apache 2.0

### PR DESCRIPTION
**Pull request by mehrdada**

### CODE REVIEW

**Comment by jtattermusch** [comment]
LGTM  for changing the licence to Apache 2.0, but we'll probably need an explicit approval from the grpc-haskell maintainers (not me).

**Comment by mehrdada** [comment]
cc/@kolmodin PTAL

File: LICENSE [hunk 1]
```diff
@@ -1,28 +1,202 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright [yyyy] [name of copyright owner]
```
Comment (kolmodin) [inline LICENSE]: Should this be filled in?

File: grpc.cabal [hunk 2]
```diff
@@ -4,12 +4,12 @@ name:                grpc
 version:             0.1.0.0
 synopsis:            gRPC for Haskell
 description:         gRPC for Haskell built on the C library.
-homepage:            http://www.grpc.io
-license:             BSD3
+homepage:            https://grpc.io
+license:             OtherLicense
```
Comment (kolmodin) [inline grpc.cabal]: Please change to Apache-2.0

File: grpc.cabal.template [hunk 3]
```diff
@@ -6,12 +6,12 @@
   version:             0.1.0.0
   synopsis:            gRPC for Haskell
   description:         gRPC for Haskell built on the C library.
-  homepage:            http://www.grpc.io
-  license:             BSD3
+  homepage:            https://grpc.io
+  license:             OtherLicense
```
Comment (kolmodin) [inline grpc.cabal.template]: Please change to Apache-2.0

File: LICENSE (hunk 1 above)
Comment (mehrdada) [reply LICENSE]: Short answer: no. This is a verbatim copy of the license that describes how to use it on your own code. The actual application is on top of the source files. This is how we do it in other grpc repos too.

File: grpc.cabal.template (hunk 3 above)
Comment (mehrdada) [reply grpc.cabal.template]: Done. PTAL.

File: LICENSE (hunk 1 above)
Comment (kolmodin) [reply LICENSE]: Ok, thanks.

**Comment by kolmodin** [comment]
LGTM, thanks guys.

**Pull request merged**